torch/backends/mkldnn/__init__.py

```diff
@@ -1,6 +1,50 @@
+import sys
 import torch
+import types
+from contextlib import contextmanager
 
 
 def is_available():
     r"""Returns whether PyTorch is built with MKL-DNN support."""
     return torch._C.has_mkldnn
+
+def set_flags(_enabled):
+    orig_flags = (torch._C._get_mkldnn_enabled(),)
+    torch._C._set_mkldnn_enabled(_enabled)
+    return orig_flags
+
+@contextmanager
+def flags(enabled=False):
+    orig_flags = set_flags(enabled)
+    try:
+        yield
+    finally:
+        set_flags(orig_flags[0])
+
+# from cudnn/__init__.py
+#
+#
+class ContextProp(object):
+    def __init__(self, getter, setter):
+        self.getter = getter
+        self.setter = setter
+
+    def __get__(self, obj, objtype):
+        return self.getter()
+
+    def __set__(self, obj, val):
+        self.setter(val)
```

Comment:
This would be a good opportunity to put this somewhere common so it can be reused by both modules.